db/controllers.js

```diff
@@ -9,4 +10,25 @@ function getAllTopics(req, res) {
     })
 }
 
-module.exports = {getAllTopics};
\ No newline at end of file
+function updateEndpointsDynamically(data) {
+    return fs.writeFile("/home/hashim/northcoders/backend/be-nc-news/endpoints.json", JSON.stringify(data, null, 2), "utf-8");
+}
+
+function getAllApi(req, res) {
+    fs.readFile("/home/hashim/northcoders/backend/be-nc-news/endpoints.json", "utf-8").then((response) => {
+        const endpoints = JSON.parse(response);
+        return models.fetchAllApi().then((response) => {
+            const newEndpoints = {
+                ...endpoints,
+                ...response,
+            }
+            return updateEndpointsDynamically(newEndpoints).then((response) => {
+                res.status(200).send({newEndpoints});
+            })
+        })
+    }).catch((error) => {
+        console.error("error reading file getAllApi Func", error);
+    })
+}
```

Comment:
This is far too elaborate than what it needs to be. All we need to do is read the endpoints.json file. As it is a get request we should not be writing anything.
Needs a refactor